autoemulate/experimental/tuner.py

```diff
@@ -57,15 +59,31 @@ def run(self, model_class: type[Emulator]) -> tuple[list[float], list[ModelConfi
             }
             if isinstance(model_class, gpytorch.models.ExactGP):
                 m = model_class(train_x, train_y, **params_sample)
+                assert isinstance(m, Emulator)
                 m.fit(train_x, train_y)
             else:
                 m = model_class(**params_sample)
+                assert isinstance(m, Emulator)
                 # TODO: check if pass as dataloader
                 m.fit(train_x, train_y)
 
             # evaluate
             y_pred = m.predict(val_x)
-            score = self.score_f(val_y, y_pred.detach().numpy())
+            # handle types
+            if isinstance(y_pred, TensorLike):
```

Comment:
Just to flag, eventually I suspect this score handling will move somewhere else given we will also want to use it inside Autoemulate.compare(), but no need to decide that here.